prerequisites:

dev-libs/boost[static-libs]
>=dev-libs/protobuf-3[static-libs]
net-libs/libmicrohttpd
virtual/libusb
dev-util/cmake
net-misc/curl
virtual/pkgconfig

build instructions:

git clone https://github.com/trezor/trezord 
cd trezord
git submodule update --init
cd src/config/
wget https://raw.githubusercontent.com/trezor/trezor-common/master/protob/config.proto
patch -p3 <trezord-protobuf3.patch
protoc -I/usr/include -I. --cpp_out=. config.proto
cd ../..
./build.sh 

--- src/config/config.proto     2017-11-28 13:10:43.910590225 -0800
+++ src/config/config.proto     2017-11-28 13:09:03.964398143 -0800
@@ -2,6 +2,8 @@
  * Configuration format for TREZOR plugin
  */

+syntax = "proto2";
+
 // Sugar for easier handling in Java
 option java_package = "com.satoshilabs.trezor.lib.protobuf";
 option java_outer_classname = "TrezorConfig";
